ambry-rest/src/main/java/com.github.ambry.rest/NettyRequest.java

```diff
@@ -215,6 +222,9 @@ public void close() {
       } finally {
         contentLock.unlock();
         restRequestMetricsTracker.recordMetrics();
+        if (digestCalculationTimeInMs >= 0) {
```

Comment:
seems like this could be done within `getDigest()` and that is probably the right place? You wouldn't have to do this check either.